poshc2/server/Payloads.py

```diff
@@ -238,112 +167,107 @@ def CreateDroppers(self, name=""):
+            with open(f"{self.BaseDirectory}{DestinationFileB64}", 'wb') as b64:
                 b64.write(base64.b64encode(binary.read()))
 
-        # Load CLR "v4.0.30319"
+    def CreateDlls(self, name=""):
         self.QuickstartLog(Colours.END)
-        self.QuickstartLog("Shellcode that loads CLR v4.0.30319" + Colours.GREEN)
-        with open('%sPosh_v4_x86_Shellcode.b64' % PayloadTemplatesDirectory, 'rb') as f:
-            v4_86 = f.read()
-        self.PatchBytes("%sPosh_v4_x86_Shellcode.bin" % name, v4_86, offsetFinder('%sPosh_v4_x86_Shellcode.b64' % PayloadTemplatesDirectory), "Shellcode")
-        with open("%s%sPosh_v4_x86_Shellcode.bin" % (self.BaseDirectory, name), 'rb') as binary:
-            with open("%s%sPosh_v4_x86_Shellcode.b64" % (self.BaseDirectory, name), 'wb') as b64:
-                b64.write(base64.b64encode(binary.read()))
-        with open('%sPosh_v4_x64_Shellcode.b64' % PayloadTemplatesDirectory, 'rb') as f:
-            v4_64 = f.read()
-        self.PatchBytes("%sPosh_v4_x64_Shellcode.bin" % name, v4_64, offsetFinder('%sPosh_v4_x64_Shellcode.b64' % PayloadTemplatesDirectory), "Shellcode")
-        with open("%s%sPosh_v4_x64_Shellcode.bin" % (self.BaseDirectory, name), 'rb') as binary:
-            with open("%s%sPosh_v4_x64_Shellcode.b64" % (self.BaseDirectory, name), 'wb') as b64:
-                b64.write(base64.b64encode(binary.read()))
+        self.QuickstartLog("ReflectiveDLL that loads CLR v2.0.50727 - DLL Export (VoidFunc)" + Colours.GREEN)
+        self.CreateDll(f"{name}Posh_v2_x86.dll", f"{PayloadTemplatesDirectory}Posh_v2_x86_dll.b64", "DLL")
+        self.CreateDll(f"{name}Posh_v2_x64.dll", f"{PayloadTemplatesDirectory}Posh_v2_x64_dll.b64", "DLL")
 
-        # Load CLR "v4.0.30319" via SharpDLL
-        with open('%sSharp_v4_x86_Shellcode.b64' % PayloadTemplatesDirectory) as f:
-            v4_86 = f.read()
-        self.PatchSharpBytes("%sSharp_v4_x86_Shellcode.bin" % name, v4_86, offsetFinder('%sSharp_v4_x86_Shellcode.b64' % PayloadTemplatesDirectory), name)
-        with open("%s%sSharp_v4_x86_Shellcode.bin" % (self.BaseDirectory, name), 'rb') as binary:
-            with open("%s%sSharp_v4_x86_Shellcode.b64" % (self.BaseDirectory, name), 'wb') as b64:
-                b64.write(base64.b64encode(binary.read()))
-        with open('%sSharp_v4_x64_Shellcode.b64' % PayloadTemplatesDirectory) as f:
-            v4_64 = f.read()
-        self.PatchSharpBytes("%sSharp_v4_x64_Shellcode.bin" % name, v4_64, offsetFinder('%sSharp_v4_x64_Shellcode.b64' % PayloadTemplatesDirectory), name)
-        with open("%s%sSharp_v4_x64_Shellcode.bin" % (self.BaseDirectory, name), 'rb') as binary:
-            with open("%s%sSharp_v4_x64_Shellcode.b64" % (self.BaseDirectory, name), 'wb') as b64:
-                b64.write(base64.b64encode(binary.read()))
+        self.QuickstartLog(Colours.END)
+        self.QuickstartLog("ReflectiveDLL that loads CLR v4.0.30319 - DLL Export (VoidFunc)" + Colours.GREEN)
```

Comment:
How come we hard code the name to "DLL"?

Reply:
Yeah that was a bit bizzare, fixed that